### PR TITLE
[WIP] Enviroment var config

### PR DESCRIPTION
**Pull request by deg4uss3r**
Work in progress PR for #375 

Happy to change as you see fit, tested manually with a --style=numbers line, looking for a good way to write an integration test for the config `$VAR`

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -9,7 +9,17 @@ use dirs::PROJECT_DIRS;
 use util::transpose;
 
 pub fn config_file() -> PathBuf {
-    PROJECT_DIRS.config_dir().join("config")
+    match env::var("BAT_CONFIG_PATH") {
```
Comment (sharkdp) [inline src/config.rs]: An alternative way to write this logic would be with the combinators on `Option`:
```rust
env::var("BAT_CONFIG_PATH")
    .ok()
    .map(PathBuf::from)
    .filter(Path::is_file)
    .unwrap_or(PROJECT_DIRS.config_dir().join("config"))
```

**Comment by sharkdp** [comment]
Looks good, thank you very much for your contribution!

**Comment by sharkdp** [comment]
If you want to write an integration test for this (which would be great), you could take a look at the `pager_basic` test in `integration_tests.rs` for an example on how to set an environment variable.

You would probably have to add an exemplary config file somewhere in the `tests` folder and set the `BAT_CONFIG_PATH` variable to that file.

File: src/config.rs (hunk 1 above)
Comment (deg4uss3r) [reply src/config.rs]: FYI I am working on this looks like below, but seems to be causing some weird issues with my build. I'll keep at it :) 

```rust
env::var("BAT_CONFIG_PATH")
        .ok()
        .map(PathBuf::from)
        .filter(|config_path| config_path.is_file())
        .unwrap_or(PROJECT_DIRS.config_dir().join("config"))
```

Comment (sharkdp) [reply src/config.rs]: Okay, let me know if you need help.

Comment (deg4uss3r) [reply src/config.rs]: Added two integration tests, one to check if it's reading the environment variable path right, and another that enables paging and checks for the paging output like `pager_basic()`

**Comment by sharkdp** [comment]
Thank you very much for the updates. I made some small modifications on top of yours.

**Comment by deg4uss3r** [comment]
Cool, these make sense! Is there anything else you would like me to add? I can type up an addition to the README if you’d like.

**Comment by sharkdp** [comment]
> I can type up an addition to the README if you’d like.

Oh, that would be great! Unfortunately, there is no section about the config file, yet (see #377).

**Comment by deg4uss3r** [comment]
> see #377

Sure! I will open a new PR for that. First commit will be very basic to discuss how it should be formatted and getting all the options included. Is that okay?

**Comment by sharkdp** [comment]
> Sure! I will open a new PR for that. First commit will be very basic to discuss how it should be formatted and getting all the options included. Is that okay?

Sounds great.

**Pull request merged**